app/src/main/java/com/seafile/seadroid2/data/DatabaseHelper.java

```diff
@@ -518,4 +547,56 @@ public int getCachedDirentUsage(String dirID) {
         return count;
     }
 
+    public Pair<String, String> getEnckey(@NonNull String repoId) {
+        String[] projection = {
+                ENCKEY_COLUMN_ENCKEY,
+                ENCKEY_COLUMN_ENCIV
+        };
+
+        String selectClause = String.format("%s = ?",
+                ENCKEY_COLUMN_REPO_ID);
+
+        String [] selectArgs = { repoId };
+
+        Cursor cursor = database.query(
+                        ENCKEY_TABLE_NAME,
+                        projection,
+                        selectClause,
+                        selectArgs,
+                        null,   // don't group the rows
+                        null,   // don't filter by row groups
+                        null);  // The sort order
+
+        if (!cursor.moveToFirst()) {
+            cursor.close();
+            return null;
+        }
+
+        return new Pair<>(cursor.getString(0), cursor.getString(1));
+    }
+
+    public void saveEncKey(@NonNull String encKey, @NonNull String encIv, @NonNull String repoId) {
+        Pair<String, String> old = getEnckey(repoId);
+
+        if (old != null && !TextUtils.isEmpty(old.first)) {
+            if (old.first.equals(encKey) && old.second.equals(encIv)) {
+                return;
+            } else {
+                delEnckey(repoId);
+            }
+        }
+
+        // Create a new map of values, where column names are the keys
+        ContentValues values = new ContentValues();
+        values.put(ENCKEY_COLUMN_ENCKEY, encKey);
+        values.put(ENCKEY_COLUMN_ENCIV, encIv);
+        values.put(ENCKEY_COLUMN_REPO_ID, repoId);
+
+        database.insert(ENCKEY_TABLE_NAME, null, values);
+    }
+
+    private void delEnckey(String repoId) {
```

Comment:
1. enc_keys will keep valid forever
2. if user turned off the enc settings, use password by default, valid period is 1 hour
3. it supports delete database entries now

> from a UI point of view, does the user understand that the Android device will store the secret forever?
Not implemented yet.